spec/chartmogul/api_resource_spec.rb

```diff
@@ -3,19 +3,42 @@
 require 'spec_helper'
 
 describe ChartMogul::APIResource do
-  describe 'connection', vcr: true do
-    it 'works when credentials are updated', vcr: { record: :all } do
+  before do
+    WebMock.allow_net_connect!
+  end
+  after do
+    WebMock.disable_net_connect!
+  end
+
+  describe 'connection' do
+    it 'works when credentials are updated', vcr: { record: :all, exclusive: true } do
       set_invalid_credentials
       expect { ChartMogul::Ping.ping }.to raise_error(ChartMogul::UnauthorizedError)
 
       set_valid_credentials
-      expect { ChartMogul::Ping.ping }.not_to raise_error(ChartMogul::UnauthorizedError)
+      expect { ChartMogul::Ping.ping }.not_to raise_error
+    end
+
+    it 'works in a threaded environment', vcr: { record: :all, exclusive: true } do
+      t1 = Thread.new do
```

Comment:
This is starting two new threads and joining into the main one later. The order of their execution on a 1-core CPU is random, to make it a bit more predictable, it's better to set something in the main thread and then run another thread (join).